resources/js/components/forms/album/AlbumVisibility.vue

```diff
@@ -0,0 +1,148 @@
+						<p class="mt-1.5 mb-4">{{ $t("lychee.ALBUM_PASSWORD_PROT_EXPL") }}</p>
+						<FloatLabel v-if="is_password_required">
+							<InputPassword id="password" v-model="password" autocomplete="new-password" @change="save" />
+							<label for="password">{{ $t("lychee.ALBUM_PASSWORD") }}</label>
+						</FloatLabel>
+					</div>
+				</Collapse>
+			</form>
+			<template v-if="props.config.is_base_album">
+				<hr class="block mt-8 mb-8 w-full border-t border-solid border-surface-600" />
+				<form>
+					<div class="relative h-12 my-4 transition-color duration-300">
+						<ToggleSwitch
+							id="pp_dialog_nsfw_check"
+							v-model="is_nsfw"
+							class="mr-2 translate-y-1"
+							style="
+								--p-toggleswitch-checked-background: var(--p-red-800);
+								--p-toggleswitch-checked-hover-background: var(--p-red-900);
+								--p-toggleswitch-hover-background: var(--p-red-900);
+							"
+							@change="save"
+						/>
+						<label for="pp_dialog_nsfw_check" class="font-bold" :class="is_nsfw ? ' text-red-700' : ''">{{
+							$t("lychee.ALBUM_NSFW")
+						}}</label>
+						<p class="my-1.5">{{ $t("lychee.ALBUM_NSFW_EXPL") }}</p>
+					</div>
+				</form>
+			</template>
+		</template>
+	</Card>
+</template>
+<script setup lang="ts">
+import { ref } from "vue";
+import Card from "primevue/card";
+import ToggleSwitch from "primevue/toggleswitch";
+import FloatLabel from "primevue/floatlabel";
+import InputPassword from "@/components/forms/basic/InputPassword.vue";
+import AlbumService, { UpdateProtectionPolicyData } from "@/services/album-service";
+import { useToast } from "primevue/usetoast";
+import { Collapse } from "vue-collapsed";
+
+const props = defineProps<{
+	album: App.Http.Resources.Models.AlbumResource | App.Http.Resources.Models.SmartAlbumResource | App.Http.Resources.Models.TagAlbumResource;
+	config: App.Http.Resources.GalleryConfigs.AlbumConfig;
+}>();
+const toast = useToast();
+const albumId = ref<string>(props.album.id);
+const is_public = ref<boolean>(props.album.policy.is_public);
+const is_link_required = ref<boolean>(props.album.policy.is_link_required);
+const is_nsfw = ref<boolean>(props.album.policy.is_nsfw);
+const grants_full_photo_access = ref<boolean>(props.album.policy.grants_full_photo_access);
+const grants_download = ref<boolean>(props.album.policy.grants_download);
+const is_password_required = ref<boolean>(props.album.policy.is_password_required);
+const password = ref<string>("");
+
+function save() {
+	const data: UpdateProtectionPolicyData = {
+		album_id: albumId.value,
+		is_public: is_public.value,
+		is_link_required: is_link_required.value,
+		is_nsfw: is_nsfw.value,
+		grants_full_photo_access: grants_full_photo_access.value,
+		grants_download: grants_download.value,
+		password: password.value,
+	};
+
+	AlbumService.updateProtectionPolicy(data)
+		.then(() => {
+			toast.add({ severity: "success", summary: "Success", detail: "Permission deleted", life: 3000 });
```

Comment:
This message makes some sense as a developer, but I can see it concerning users (I had the thought when playing around with the alpha, then saw it here)

Even if it's just
```suggestion
			toast.add({ severity: "success", summary: "Success", detail: "Permission updated", life: 3000 });
```
?

Reply:
Looks like this is fixed in #2567

Reply:
yep. :)